### PR TITLE
chore(release): release v0.9.0-rc.1

### PR DESCRIPTION
**Pull request by allenlsy**
Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

Create release v0.9.0-rc.1

- [x] I have cherry-picked any changes [labelled](https://github.com/openservicemesh/osm/labels) `needs-cherry-pick-vX.Y.Z`
- [x] I have made all of the following version and patch updates:
  1. Updated the container image tag in charts/osm/values.yaml
  2. Updated the chart **and** app version in charts/osm/Chart.yaml
  3. Updated the default osm-controller image tag in osm cli
  4. Updated the image tags used in the demo manifests
  5. Regenerated the Helm chart README.md

- [x] I have checked that the base branch for this PR is correct as defined by the release guide

Is this a release candidate?

Yes

### CODE REVIEW

File: charts/osm/Chart.yaml [hunk 1]
```diff
@@ -14,11 +14,11 @@ type: application
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 0.8.4
+version: 0.9.0
```
Comment (nojnhuh) [inline charts/osm/Chart.yaml:17]: Same for all instances:
```suggestion
version: 0.9.0-rc.1
```

**Review by nojnhuh — CHANGES_REQUESTED** [review]
Need to update versions to `v0.9.0-rc.1`

File: docs/release_guide.md [hunk 2]
```diff
@@ -52,7 +52,6 @@ Create a new commit on the new branch to update the hardcoded version informatio
 
 * The container image tag in [charts/osm/values.yaml](https://github.com/openservicemesh/osm/tree/release-v0.8/charts/osm/values.yaml)
 * The chart and app version in [charts/osm/Chart.yaml](https://github.com/openservicemesh/osm/tree/release-v0.8/charts/osm/Chart.yaml)
-* The default osm-controller image tag in [osm cli install](https://github.com/openservicemesh/osm/blob/release-v0.8/cmd/cli/install.go)
```
Comment (snehachhabria) [inline docs/release_guide.md:55]: nit: deletion of this should not be a part of this pr

File: charts/osm/Chart.yaml (hunk 1 above)
Comment (allenlsy) [reply charts/osm/Chart.yaml]: Done.

File: docs/release_guide.md (hunk 2 above)
Comment (allenlsy) [reply docs/release_guide.md]: Reverted

**Review by snehachhabria — APPROVED** [review]

**Comment by shashankram** [comment]
> Signed-off-by: Allen Leigh [allenlsy@gmail.com](mailto:allenlsy@gmail.com)
> 
> **Description**:
> 
> Create release v0.9.0-rc.1
> 
> * [ ]  I have cherry-picked any changes [labelled](https://github.com/openservicemesh/osm/labels) `needs-cherry-pick-vX.Y.Z`
> * [ ]  I have made all of the following version and patch updates:
>   
>   1. Updated the container image tag in charts/osm/values.yaml
>   2. Updated the chart **and** app version in charts/osm/Chart.yaml
>   3. Updated the default osm-controller image tag in osm cli
>   4. Updated the image tags used in the demo manifests
>   5. Regenerated the Helm chart README.md
> * [ ]  I have checked that the base branch for this PR is correct as defined by the release guide
> 
> Is this a release candidate?
> 
> Yes

@allenlsy Kindly fill up the PR template.

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**